nixos/modules/services/web-apps/firefly-iii.nix

```diff
@@ -0,0 +1,356 @@
+{ pkgs, config, lib, ... }:
+
+let
+  inherit (lib) mkIf mkOption mkEnableOption literalExpression mdDoc;
+  inherit (lib.types) attrsOf oneOf str int bool path package enum submodule;
+
+  cfg = config.services.firefly-iii;
+  format = pkgs.formats.ini { };
+
+  user = cfg.user;
+  group = cfg.group;
+
+  defaultUser = "firefly-iii";
+  defaultGroup = defaultUser;
+
+  firefly-pkg = pkgs.firefly-iii.override {
+    dataDir = cfg.dataDir;
+    settings = cfg.settings;
+  };
+
+  firefly-php = "${cfg.package.phpPackage}/bin/php";
+  artisan = "${cfg.package}/artisan";
+
+  fileenv-func = ''
+    file_env() {
```

Comment:
Any chance we can simplify this?

Reply:
This is taken from [upstream](https://dev.azure.com/Firefly-III/_git/MainImage?path=%2Fentrypoint-fpm.sh). We probably can simplify it, but it seems to be a widely used function in the docker world.

Reply:
Ah, if it is just copied from wordpress, then that's fine for me.